build fail looks real. I don't feel like I have enough context to review so will let @gcapalbo continue to own.
```
======================================================================
FAIL: test_twilio_domain_level_backend (corehq.apps.smsbillables.tests.test_gateway_fees.TestGatewayFee)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/corehq/apps/smsbillables/tests/test_gateway_fees.py", line 322, in test_twilio_domain_level_backend
    self.assertEqual(mock_log_smsbillables_error.call_count, 0)
AssertionError: 20 != 0
----------------------------------------------------------------------
Ran 150 tests in 165.412s
``

--- a/corehq/apps/sms/tasks.py
+++ b/corehq/apps/sms/tasks.py
@@ -13,6 +13,7 @@ from django.conf import settings
 from corehq import privileges
 from corehq.apps.accounting.utils import domain_has_privilege
 from corehq.apps.domain.models import Domain
+from corehq.apps.smsbillables.exceptions import RetryBillableTaskException
 from corehq.apps.smsbillables.models import SmsBillable
 from corehq.apps.sms.change_publishers import publish_sms_saved
 from corehq.util.timezones.conversions import ServerTime
@@ -238,8 +239,8 @@ def process_sms(queued_sms_pk):
             process_sms.delay(queued_sms_pk)
 
 
-@task(ignore_result=True)
-def store_billable(msg):
+@task(ignore_result=True, default_retry_delay=5 * 60, max_retries=10, bind=True)
+def store_billable(self, msg):
     if not isinstance(msg, SMS):
         raise Exception("Expected msg to be an SMS")
 
@@ -251,10 +252,13 @@ def store_billable(msg):
             # This string contains unicode characters, so the allowed
             # per-sms message length is shortened
             msg_length = 70
-        SmsBillable.create(
-            msg,
-            multipart_count=int(math.ceil(float(len(msg.text)) / msg_length)),
-        )
+        try:
+            SmsBillable.create(
+                msg,
+                multipart_count=int(math.ceil(float(len(msg.text)) / msg_length)),
+            )
+        except RetryBillableTaskException as e:
+            self.retry(exc=e)
 
 
 @task(queue='background_queue', ignore_result=True, acks_late=True)

--- a/corehq/apps/smsbillables/exceptions.py
+++ b/corehq/apps/smsbillables/exceptions.py
@@ -4,3 +4,7 @@ class AmbiguousPrefixException(Exception):
 
 class SMSRateCalculatorError(Exception):
     pass
+
+
+class RetryBillableTaskException(Exception):
+    pass

--- a/corehq/apps/smsbillables/models.py
+++ b/corehq/apps/smsbillables/models.py
@@ -5,6 +5,7 @@ from django.conf import settings
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import models
 
+from twilio import TwilioRestException
 from twilio.rest import TwilioRestClient
 
 from corehq.apps.accounting import models as accounting
@@ -20,7 +21,7 @@ from corehq.apps.sms.phonenumbers_helper import get_country_code_and_national_nu
 from corehq.apps.smsbillables.utils import log_smsbillables_error
 from corehq.messaging.smsbackends.test.models import SQLTestSMSBackend
 from corehq.apps.sms.util import clean_phone_number
-from corehq.apps.smsbillables.exceptions import AmbiguousPrefixException
+from corehq.apps.smsbillables.exceptions import AmbiguousPrefixException, RetryBillableTaskException
 from corehq.messaging.smsbackends.twilio.models import SQLTwilioBackend
 from corehq.util.quickcache import quickcache
 
@@ -407,25 +408,23 @@ class SmsBillable(models.Model):
             config = twilio_backend.config
             return TwilioRestClient(config.account_sid, config.auth_token)
 
-        try:
-            if backend_message_id:
+        if backend_message_id:
+            try:
                 twilio_message = _get_twilio_client().messages.get(backend_message_id)
-                return _TwilioChargeInfo(
-                    Decimal(twilio_message.price) * -1,
-                    SmsGatewayFee.get_by_criteria(
-                        SQLTwilioBackend.get_api_id(),
-                        direction,
-                    )
+            except TwilioRestException:
+                raise RetryBillableTaskException
+            return _TwilioChargeInfo(
+                Decimal(twilio_message.price) * -1,
+                SmsGatewayFee.get_by_criteria(
+                    SQLTwilioBackend.get_api_id(),
+                    direction,
                 )
-            else:
-                log_smsbillables_error(
-                    "Could not create gateway fee for Twilio message %s: no backend_message_id" % couch_id
-                )
-        except Exception as e:
-            log_smsbillables_error(
-                "Error looking up Twilio gateway fee for SMSLog %s: %s" % (couch_id, e.message)
             )
-        return _TwilioChargeInfo(None, None)
+        else:
+            log_smsbillables_error(
+                "Could not create gateway fee for Twilio message %s: no backend_message_id" % couch_id
+            )
+            return _TwilioChargeInfo(None, None)
 
 _TwilioChargeInfo = namedtuple('_TwilioCharges', ['twilio_gateway_fee', 'gateway_fee'])
 _GatewayChargeInfo = namedtuple('_GatewayChargeInfo', ['gateway_fee', 'conversion_rate', 'direct_gateway_fee'])

--- a/corehq/apps/smsbillables/models.py
+++ b/corehq/apps/smsbillables/models.py
@@ -298,10 +298,7 @@ class SmsBillable(models.Model):
     def _single_gateway_charge(self):
         amount = None
         if self.gateway_fee is not None:
-            try:
-                amount = SmsGatewayFee.objects.get(id=self.gateway_fee.id).amount
-            except SmsGatewayFee.DoesNotExist:
-                pass
+            amount = self.gateway_fee.amount
         if amount is None:
             amount = self.direct_gateway_fee or Decimal('0.0')
 
@@ -312,11 +309,7 @@ class SmsBillable(models.Model):
     @property
     def _single_usage_charge(self):
         if self.usage_fee is not None:
-            try:
-                charge = SmsUsageFee.objects.get(id=self.usage_fee.id)
-                return charge.amount
-            except ObjectDoesNotExist:
-                pass
+            return self.usage_fee.amount
         return Decimal('0.0')
 
     @classmethod

--- a/corehq/apps/smsbillables/tests/test_gateway_fees.py
+++ b/corehq/apps/smsbillables/tests/test_gateway_fees.py
@@ -7,6 +7,7 @@ from django.test import TestCase
 from corehq.apps.accounting.generator import init_default_currency
 from corehq.apps.sms.models import SMS, SQLMobileBackend
 from corehq.apps.smsbillables import generator
+from corehq.apps.smsbillables.management.commands.bootstrap_usage_fees import bootstrap_usage_fees
 from corehq.apps.smsbillables.models import (
     add_twilio_gateway_fee,
     SmsBillable,
@@ -290,6 +291,7 @@ class TestGatewayFee(TestCase):
     )
     def test_twilio_domain_level_backend(self, mock_log_smsbillables_error):
         add_twilio_gateway_fee(apps)
+        bootstrap_usage_fees(apps)
         twilio_backend = SQLTwilioBackend.objects.create(
             name='TWILIO',
             is_global=False,

